Assignments/07b - Writing Files/README.md

```diff
@@ -0,0 +1,96 @@
+Task(s)
+-------
+The syntax for writing to files is very similar to that of reading files. Compare the two pieces of code below.
+
+```python
+#Reading a file
+file = open('document.txt', 'r')
+fileContent = file.readlines()
+file.close()
+...
+```
+
+```python
+#Writing to a file
+file = open('document.txt', 'w')
+file.writelines("content")
+file.close()
+...
+```
+
+In the second line of both versions, the file "document.txt" is opened and stored in the variable "file". The only difference between the reading version and writing version is that when you're reading a file, you use "r", and when you're writing, you use "w". 
+
+Now that "document.txt" is stored in a variable and Python knows that we plan to be writing to it, we can tell it what to write. In the writing example, the word "content" will be written to the file. If "document.txt" already has data, it will be overwritten with just the word "content". If we want to simply add the word "content" at the very end of "document.txt" without affecting what is already in the file, we would replace the "w" in line 2 with "a", where "a" stands for *append*.
+
+Now that we are done with the file, we close it with "file.close()", for good housekeeping.
+
+
+For this assignment, you will need some extra information. Here is an example of a substitution cipher, which that decodes an encrypted text.
+
+```python
+f = open('mysteryText.txt', 'r')
+fileContent = f.readlines()
+f.close()
+
+alphabet = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
+key = 'yblavdnrwexizgqskujhcftmopYBLAVDNRWEXIZGQSKUJHCFTMOP' #'a' gets translated to 'y', 'b' gets translated to 'b', and so on.
+
+def decode(line):
+  decodedLine = ""
+  for character in line:                        #"character" is a string of length 1.
```

Comment:
They haven't used a lot of "x in y" style work in class - ensure you go over this during the lesson.